From c5526479fcc7ab4ee3734d8d662912c70433cf06 Mon Sep 17 00:00:00 2001
From: Mark Allyn <mark.a.allyn@intel.com>
Date: Mon, 26 Nov 2012 12:41:10 -0800
Subject: [PORT FROM MAIN] Phone Encryption: Data partition does not unmount

BZ: 64614

A process using the /data partition must not be shutting down in the
10 seconds allowed by the un-mount loop in cryptfs.

The loop counter is incremented twice, removing one of the increments
allows the the unmount to retry for 20 seconds which gives the process
using the data partition enough time to shutdown.

Change-Id: I7ab0ce7b0fff655b3411f0ade527991172b4c87f
Orig-Change-Id: I667130e256aa7b907f065cfae0cc18ff23e6f4ca
Reviewed-on: http://android.intel.com:8080/76378
Reviewed-by: Allyn, Mark A <mark.a.allyn@intel.com>
Reviewed-by: Shuster, Norvin <norvin.shuster@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Reviewed-by: Mangalampalli, Jayant <jayant.mangalampalli@intel.com>
Tested-by: Mangalampalli, Jayant <jayant.mangalampalli@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 cryptfs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cryptfs.c b/cryptfs.c
index d926334..c3c4d57 100644
--- a/cryptfs.c
+++ b/cryptfs.c
@@ -656,7 +656,7 @@ static int wait_and_unmount(char *mountpoint)
                 break;
             }
             sleep(1);
-            i++;
+
         } else {
           break;
         }
-- 
1.7.4.1

